logging/alternateServiceLogger.go

```diff
@@ -52,16 +54,46 @@ func NewZapSugaredServiceLogger(lc AltServiceLoggerConfig) AltServiceLogger {
 	config.EncoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
 	config.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
 
-	zapLogger, err := config.Build(zap.AddCallerSkip(lc.SkipLevels))
+	var withCaller bool = true
+	if lc.SkipLevels == 1 {
+		withCaller = false
+	}
+	zapLogger, err := config.Build(zap.WithCaller(withCaller))
 	if err != nil {
 		panic(fmt.Errorf("Logger could not be built. This is not an expected outcome. ERR: %+v", err))
 	}
+
+	var loggingFormat = utils.MustGetenv("LOGGING_FORMAT")
+	var enc zapcore.Encoder
+	if strings.EqualFold("Syslog", loggingFormat) {
+		enc = NewSyslogEncoder(SyslogEncoderConfig{
+			EncoderConfig: zapcore.EncoderConfig{
+				NameKey:        "logger",
+				CallerKey:      "caller",
+				MessageKey:     "msg",
+				StacktraceKey:  "stacktrace",
+				EncodeLevel:    zapcore.LowercaseLevelEncoder,
+				EncodeTime:     zapcore.EpochTimeEncoder,
+				EncodeDuration: zapcore.SecondsDurationEncoder,
+				//EncodeCaller:   zapcore.ShortCallerEncoder,
+			},
+
+			Facility:  LOG_LOCAL0,
+			Hostname:  "localhost",
```

Comment:
should this be an arg, incase its prod vs staging?

Reply:
Yes katie. I will update this to be get from Env file.